reader_concurrency_semaphore.cc

```diff
@@ -304,19 +305,34 @@ class reader_permit::impl
     }
 
     db::timeout_clock::time_point timeout() const noexcept {
-        return _timeout;
+        return _timeout_timer.armed() ? _timeout_timer.get_timeout() : db::no_timeout;
```

Comment:
Do we still need this function?

Reply:
I guess we do. Perhaps we can/should make it private, to ensure everyone is using abort_source rather than checking timeouts manually?

Reply:
We still use the `timeout()` in the multishard reader to create a permit with the same timeout on the remote shard.